addons/video/video.c

```diff
@@ -137,6 +138,42 @@ ALLEGRO_VIDEO *al_open_video(char const *filename)
    return video;
 }
 
+ALLEGRO_VIDEO* al_open_video_f(ALLEGRO_FILE* fp)
```

Comment:
Please also add a `, const char *ident`, as is done for, e.g., `al_load_bitmap_f`.

The logic below would then be changed to something like:

```c
if (!ident)
   ident = al_identify_video_f(fp);
```